### PR TITLE
Fix compiler warnings in KQIR and RDB

### PR DESCRIPTION
**Pull request by raffertyyu**
According to gcc output, fix two compiler warnings.

### CODE REVIEW

File: src/storage/rdb_ziplist.cc [hunk 1]
```diff
@@ -196,7 +196,7 @@ uint32_t ZipList::ZipStoreEntryEncoding(unsigned char *p, size_t zl_size, unsign
     buf[3] = (rawlen >> 8) & 0xff;
     buf[4] = rawlen & 0xff;
   }
-  assert(zl_size >= zlHeaderSize + len);
+  assert(zl_size >= static_cast<size_t>(zlHeaderSize) + len);
```
Comment (mapleFU) [inline src/storage/rdb_ziplist.cc:199]: Would `DCHECK` better here?

File: src/search/executors/filter_executor.h [hunk 2]
```diff
@@ -100,6 +100,9 @@ struct QueryExprEvaluator {
         return l > r;
       case NumericCompareExpr::GET:
         return l >= r;
+      default:
+        assert(false);
+        return false;
```
Comment (mapleFU) [inline src/search/executors/filter_executor.h]: Should we elaborate all branches rather than a unreachable "default"

Comment (raffertyyu) [reply src/search/executors/filter_executor.h]: Actually all branches are covered here, but gcc thought it should have a default branch to cover any unforeseen case.

File: src/storage/rdb_ziplist.cc (hunk 1 above)
Comment (raffertyyu) [reply src/storage/rdb_ziplist.cc:199]: static_cast is enough here, imo

File: src/search/executors/filter_executor.h (hunk 2 above)
Comment (PragmaTwice) [inline src/search/executors/filter_executor.h]:
```suggestion
      default:
        CHECK(false) << "unreachable";
        __builtin_unreachable();
```

**Review by PragmaTwice — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
The CI failure is not related to this PR, will take a look later.

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**